modules/registers.xql

```diff
@@ -171,15 +152,21 @@ declare function rapi:next($type) {
     switch ($type)
         case 'place'
             return collection($config:register-root)/id($config?id)//tei:place[starts-with(@xml:id, $config?prefix)]/substring-after(@xml:id, $config?prefix)
+        case 'organization'
+            return collection($config:register-root)/id($config?id)//tei:org[starts-with(@xml:id, $config?prefix)]/substring-after(@xml:id, $config?prefix)
+        case 'term'
+            return collection($config:register-root)/id($config?id)//tei:category[starts-with(@xml:id, $config?prefix)]/substring-after(@xml:id, $config?prefix)
+        case 'work'
+            return collection($config:register-root)/id($config?id)//(tei:bibl|tei:biblStruct)[starts-with(@xml:id, $config?prefix)]/substring-after(@xml:id, $config?prefix)
         default 
             return collection($config:register-root)/id($config?id)//tei:person[starts-with(@xml:id, $config?prefix)]/substring-after(@xml:id, $config?prefix)
-    
-    let $last := if (count($all-ids)) then sort($all-ids)[last()] else 1
+    let $numeric-ids := for $id in $all-ids return if ($id castable as xs:integer) then $id else ()
+    let $last := if (count($all-ids)) then sort($numeric-ids)[last()] else 1
     let $next :=
             try {
                 xs:integer($last) + 1
             } catch * {
-                '_error'
+                1
```

Comment:
These changes here are due to a case in which the register already has entries like: `rp-gnd-118536109` (in which `rp-` is the prefix) and then we want to generate custom entries with an ID like `rp-000001`